### PR TITLE
Feature/sync local recipes with firebase

### LINKED ISSUE #14: [Feature] Sync local recipes with Firestore
**Is your feature request related to a problem? Please describe.**
Currently, if a user changes its device it loses all local recipes.

**Describe your solution**
The local recipes should be stored in a remote database and when the user changes devices the application should download all the user's recipes to the new device.

### PR DESCRIPTION
**Pull request by AlbertPG**
#### Summary of what has been done *

- Add sync between local data and firebase

#### Reference issues *

Closes #14 

## Type of change *
- [x] New feature (enhancement)
- [ ] Bug fix
- [ ] Documentation update  
- [ ] Other (build scripts, metadata, assets, etc.)

## Checklist *
- [x] I have performed a self-review of my own code
- [x] My code is self-explanatory  
- [ ] I have added all new string resources into strings.xml in all languages
- [x] All possible errors are properly handled  
- [x] I have tested the application with my changes and verify that all tests are passing
- [ ] The documentation is up to date

## Testing notes

You should change the hardcoded uid to any that has not been used in firebase yet

### CODE REVIEW

File: app/src/main/res/values-ca/arrays.xml [hunk 1]
```diff
@@ -23,6 +23,6 @@
     </string-array>
     <string-array name="remote_languages">
         <item>Anglès</item>
-        <!--        <item>Espanyol (beta)</item>-->
+        <item>Espanyol (beta)</item>
```
Comment (santidrj) [inline app/src/main/res/values-ca/arrays.xml]: It seems that the API in Spanish is no longer available, so we should remove this option from the configuration and remote recipe search.

File: app/src/main/res/values-es/arrays.xml [hunk 2]
```diff
@@ -23,6 +23,6 @@
     </string-array>
     <string-array name="remote_languages">
         <item>Inglés</item>
-        <!--        <item>Español (beta)</item>-->
+        <item>Español (beta)</item>
```
Comment (santidrj) [inline app/src/main/res/values-es/arrays.xml]: It seems that the API in Spanish is no longer available, so we should remove this option from the configuration and remote recipe search

File: app/src/main/res/values/arrays.xml [hunk 3]
```diff
@@ -28,10 +28,10 @@
     </string-array>
     <string-array name="remote_languages">
         <item>English</item>
-        <!--        <item>Spanish (beta)</item>-->
+        <item>Spanish (beta)</item>
     </string-array>
     <string-array name="remote_languages_values">
         <item>en</item>
-        <!--        <item>es</item>-->
+        <item>es</item>
```
Comment (santidrj) [inline app/src/main/res/values/arrays.xml]: It seems that the API in Spanish is no longer available, so we should remove this option from the configuration and remote recipe search

**Review by santidrj — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-ca/arrays.xml (hunk 1 above)
Comment (AlbertPG) [reply app/src/main/res/values-ca/arrays.xml]: I have commented this line again. In order to remove Spanish from available remote languages, we should create a new issue.

File: app/src/main/res/values/arrays.xml (hunk 3 above)
Comment (AlbertPG) [reply app/src/main/res/values/arrays.xml]: I have commented this line again. In order to remove Spanish from available remote languages, we should create a new issue.

File: app/src/main/res/values-es/arrays.xml (hunk 2 above)
Comment (AlbertPG) [reply app/src/main/res/values-es/arrays.xml]: I have commented this line again. In order to remove Spanish from available remote languages, we should create a new issue.

**Review by santidrj — APPROVED** [review]
Good work!

**Pull request merged**